### ISSUE TITLE
Split IOutboxStorage into 2 interfaces

### ISSUE THREAD

**Issue by synhershko** [Particular/NServiceBus#2659]
In https://github.com/Particular/NServiceBus.RavenDB/pull/43#issuecomment-69909498 we've stumbled upon a possible friction point with external storages, as one method (`Store`) requires a shared session (via session provider) and the rest create their own separate sessions.

This creates a friction point as the persister requires both a session provider and a DB connection (DocumentStore in RavenDB's case), and there are some edge cases where the 2 may point to different DBs.

To avoid this, @andreasohlund suggested we split IOutboxStorage into 2 - the deduplicator side of things (`Store`) into one and the rest into another.

**Comment by SzymonPobiega** [comment]
:+1: I notices the same when working with NHibernate one

**Comment by danielmarbach** [comment]
Is this still on the radar?

**Comment by andreasohlund** [comment]
Think it's a good change, not sure about how urgent it is though

**Comment by SzymonPobiega** [comment]
I like the change. It would even more important if we decide to go for explicit context sharing between handlers because that context would only need to access the store method.

**Comment by andreasohlund** [comment]
I guess we would then pass a "context" to the `Store` to avoid needing DI.

I'm sold :)

**Comment by danielmarbach** [comment]
@DavidBoike should we close this one because it is tracked in feature dev?

**Comment by DavidBoike** [comment]
How would that affect potential future release notes?

**Comment by danielmarbach** [comment]
Good point. Then leave it open. There seems to be a lot of redundancy going
on with feature dev and issues in this repo. But what is worse: Some things
here aren't tracked yet.

**Comment by DavidBoike** [comment]
This is not really in progress, I think it's here only because of how NSB repo was brought into Waffle. Where should this go?

**Comment by DavidBoike** [comment]
Nevermind, I found its friends. All moved back to the Backlog, and I'll be removing my face as there is no current plan to do this soon.
